src/installer/managed/Microsoft.NET.HostModel/MachO/BinaryFormat/Blobs/EmbeddedSignatureBlob.cs

```diff
@@ -0,0 +1,248 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO.MemoryMappedFiles;
+#nullable enable
+
+namespace Microsoft.NET.HostModel.MachO;
+
+/// <summary>
+/// Format based off of https://github.com/apple-oss-distributions/Security/blob/3dab46a11f45f2ffdbd70e2127cc5a8ce4a1f222/OSX/libsecurity_codesigning/lib/cscdefs.h#L23
+/// Code Signature data is always big endian / network order.
+/// The EmbeddedSignatureBlob is a SuperBlob that usually contains the CodeDirectoryBlob, RequirementsBlob, and CmsWrapperBlob.
+/// The RequirementsBlob and CmsWrapperBlob may be null if the blob is not present in the read file (usually linker signed MachO files),
+/// but will be present in newly created signatures.
+/// Optionally, it may also contain the EntitlementsBlob and DerEntitlementsBlob.
+/// </summary>
+internal sealed class EmbeddedSignatureBlob : SuperBlob
```

Comment:
This doesn't look like an inheritance relationship to me, it looks like composition. Based on the linked code above, an `EmbeddedSignatureBlob` _contains_ a superblob, but is not a "kind" of superblob. I think this is true of all superblobs.

Reply:
So is the only purpose setting the MagicField number? That still seems like overkill for inheritance. Couldn't this be an enum value that's passed in?

Reply:
I just found it helpful to have different type names for each blob. Using inheritance for implementing the Magic enum just made sense after that decision. Is there that much overhead to having the different classes?